api/migrations/env.py

```diff
@@ -16,7 +16,7 @@
 # from myapp import mymodel
 # target_metadata = mymodel.Base.metadata
 from flask import current_app
-config.set_main_option('sqlalchemy.url', current_app.config.get('SQLALCHEMY_DATABASE_URI'))
+config.set_main_option('sqlalchemy.url', current_app.config.get('SQLALCHEMY_DATABASE_URI').replace("%", "%%"))
```

Comment:
what's going on here?

Reply:
python dance because of special characters in the RDS password. `%` being a python formatter thing was causing an error (would have to go back to get specific error). but then later the `%` dance needs to be undone before configuring the migration context:

```python
url = config.get_main_option("sqlalchemy.url").replace("%%", "%")
context.configure(url=url)
```

Reply:
so sqlalchemy is doing something wrong with %? It's not something we can get around by using a different type of string formatting?

Reply:
i think it's expected behaviour: https://alembic.sqlalchemy.org/en/latest/api/config.html#alembic.config.Config.set_main_option

> A raw percent sign not part of an interpolation symbol must therefore be escaped, e.g. %%.